containers/payments/subscription/BlackFridayNavbarLink.js

```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState, useRef } from 'react';
+import differenceInDays from 'date-fns/difference_in_days';
+import PropTypes from 'prop-types';
+import { useUser, useApi, useLoading, useBlackFriday, TopNavbarLink } from 'react-components';
+import { getLastCancelledSubscription } from 'proton-shared/lib/api/payments';
+
+const ONE_MONTH = 30;
+
+const BlackFridayNavbarLink = ({ to }) => {
+    const api = useApi();
+    const [loading, withLoading] = useLoading();
+    const [user] = useUser();
+    const isBlackFriday = useBlackFriday();
+    const [isEligible, setEligibility] = useState(false);
+    const checked = useRef(false);
+
+    const check = async () => {
+        // Return the latest subscription cancellation time, return null if the user never had any subscription, 0 if the user currently has an active subscription
```

Comment:
this comment refers to `api(getLastCancelledSubscription())`? Because if the return is `null`, `const { LastSubscriptionEnd = 0 } = null` will complain, no?